### PR TITLE
Bug 2035453: [Alibaba] fix api endpoint

### PR DESCRIPTION
**Pull request by menglingwei**

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -1 +1,89 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/provider"
+	sdkerrors "github.com/aliyun/alibaba-cloud-sdk-go/sdk/errors"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+)
+
+var (
+	testCreads = `
```
Comment (JoelSpeed) [inline pkg/client/client_test.go:19]: Nit, I think this should be `testCreds`

File: pkg/client/client_test.go [hunk 2]
```diff
@@ -1 +1,89 @@
...
+	endpointMap["ap-northeast-1"] = "ecs.ap-northeast-1.aliyuncs.com"
+	endpointMap["ap-southeast-2"] = "ecs.ap-southeast-2.aliyuncs.com"
+	endpointMap["ap-southeast-3"] = "ecs.ap-southeast-3.aliyuncs.com"
+	endpointMap["ap-southeast-5"] = "ecs.ap-southeast-5.aliyuncs.com"
+	endpointMap["ap-southeast-6"] = "ecs.ap-southeast-6.aliyuncs.com"
+	endpointMap["ap-south-1"] = "ecs.ap-south-1.aliyuncs.com"
+	endpointMap["eu-west-1"] = "ecs.eu-west-1.aliyuncs.com"
+	endpointMap["me-east-1"] = "ecs.me-east-1.aliyuncs.com"
+	endpointMap["eu-central-1"] = "ecs.eu-central-1.aliyuncs.com"
+	endpointMap["cn-nanjing"] = "ecs.cn-nanjing.aliyuncs.com"
```
Comment (JoelSpeed) [inline pkg/client/client_test.go:75]: Is there an exhaustive list of these somewhere we could reference to make sure have all of the possibilities? Maybe there are constants in the SDK we can leverage here?

File: pkg/client/client_test.go [hunk 3]
```diff
@@ -1 +1,89 @@
...
+
+	os.Setenv(provider.ENVCredentialFile, f.Name())
+	defer os.Unsetenv(provider.ENVCredentialFile)
+	// use Alibaba provider initialization
+	p := provider.NewProfileProvider("default")
+	// return a valid auth credential
+	return p.Resolve()
+}
+
+func Test_EcsEndPoint(t *testing.T) {
```
Comment (JoelSpeed) [inline pkg/client/client_test.go:50]: I'm not really sure what you're trying to do with this test, as far as I can tell it isn't testing any code within the cluster api provider?

File: pkg/client/client_test.go (hunk 1 above)
Comment (menglingwei) [reply pkg/client/client_test.go:19]: I will update it.

File: pkg/client/client_test.go (hunk 2 above)
Comment (menglingwei) [reply pkg/client/client_test.go:75]: Most of our cloud products have an API named DescribeRegions.  Here is the ECS api documentation. https://www.alibabacloud.com/help/en/doc-detail/25609.htm?spm=a3c0i.23458820.2359477120.4.4c6f7d3fhtz9x6

File: pkg/client/client_test.go (hunk 3 above)
Comment (menglingwei) [reply pkg/client/client_test.go:50]: My main purpose was to verify that the endpoint was working as expected. So i add this test case. We can remove it if we have to

File: pkg/client/client_test.go (hunk 2 above)
Comment (menglingwei) [reply pkg/client/client_test.go:75]: The SDK selects the correct endpoint by default. However, the ECS TEAM updated the endpoint earlier and did not update the configuration in the SDK. Therefore, when us-east-1 region is selected, an endpoint such as ecs.cn-hangzhou.aliyuncs.com is returned

File: pkg/client/client_test.go (hunk 3 above)
Comment (JoelSpeed) [reply pkg/client/client_test.go:50]: So does this test run through each of the endpoints and attempt to connect to them? Checks that the endpoints haven't changed and that the SDK is up to date? 

IMO that doesn't seem like the kind of test we would need for this project, it seems like it is likely to break in the future through no fault of our own and that would then cause issues for us as we have to fix the test

Comment (menglingwei) [reply pkg/client/client_test.go:50]: I got it. so i will remove this test.

**Comment by menglingwei** [comment]
@JoelSpeed  I removed the test case .and only update the alibaba sdk version

**Pull request merged**